openapi.yaml

```diff
@@ -164,704 +128,833 @@ paths:
+          - type
+          - attributes
+        additionalProperties: false
+        properties:
+          type:
+            $ref: '#/definitions/PersonType'
+          attributes:
+            allOf:
+              - $ref: '#/definitions/PersonPostRequired'
+              - $ref: '#/definitions/PersonPostAttributes'
+  SupervisorAttributes:
+    type: object
+    required:
+      - osuId
+      - positionNumber
+    additionalProperties: false
     properties:
-      id:
+      osuId:
         type: string
-        description: Uniquely identifies a job record for the given person.
-        example: C12345-00
-      type:
-        $ref: '#/definitions/JobType'
-      links:
-        $ref: '#/definitions/SelfLink'
-      attributes:
-        $ref: '#/definitions/JobAttributes'
-  NewJobResourceObject:
-    properties:
-      type:
-        $ref: '#/definitions/JobType'
-      attributes:
-        $ref: '#/definitions/JobAttributes'
-  JobAttributes:
+        description: OSU ID of the supervisor for the job.
+        pattern: '^\d{9}$'
+      firstName:
+        type: string
+        description: First name of the supervisor for the job.
+        example: John
+      lastName:
+        type: string
+        description: Last name of the supervisor for the job.
+        example: Smith
+      email:
+        type: string
+        format: email
+        description: The primary email of the supervisor for the job.
+        example: john.smith@example.com
+      positionNumber:
+        type: string
+        description: Position number for the supervisor.
+        example: C12345
+      suffix:
+        type: string
+        description: Suffix of the supervisor.
+  JobPostRequired:
+    type: object
     required:
       - positionNumber
+      - employmentType
       - beginDate
-      - supervisorOsuID
-      - supervisorPositionNumber
+      - supervisor
       - effectiveDate
+  JobAttributes:
+    type: object
+    additionalProperties: false
     properties:
-      positionNumber:
-        type: string
-        description: Identifies a position
-        example: C12345
       suffix:
         type: string
-        # TODO: find a better way to describe different required fields for
-        # a given set of attributes. Suffix is required for PUT but not POST.
-        description: |
+        description: >
           Used to uniquely identify a job record for persons who have
           multiple jobs that share the same position number.
           For updating existing jobs, this field is required.
+      positionNumber:
+        type: string
+        description: Identifies a position.
+        example: C12345
+      employmentType:
+        type: string
+        enum: *employmentTypeEnums
+        description: Type of employment.
       effectiveDate:
         type: string
         format: date
         description: The date this job assignment is effective.
-        example: '2018-12-31'
+        example: '1970-1-21'
```

Comment:
Should this be YYYY-MM-DD format?

Reply:
Yup. I made that mistake in a lot of places and fixed them all.